winch/codegen/src/codegen/call.rs

```diff
@@ -364,6 +313,15 @@ impl FnCall {
         // which was allocated last.
         masm.free_stack(reserved_space);
 
+        // Drop params from value stack and calculate amount of machine stack
+        // space they consumed.
+        let mut stack_consumed = 0;
+        context.drop_last(sig.params.len_without_retptr(), |_regalloc, v| {
```

Comment:
My understanding is `v` should never be a register when we reach this code because we spilled the value stack earlier. I've tried to confirm that by adding a panic if `v` is a register and running the test suite and filetests and both passed.

Reply:
Your understanding is correct, aside from matching against `Val::Memory`, I think it'd be useful to assert for such invariant in the code (e.g. `debug_assert!(v.is_mem() || v.is_const())`, that way we'd at least have a way of identifying any issues early on if this invariant ever breaks. 

NB: `is_const()` currently doesn't exist, so if we go with this approach, it needs  to be added.